### PR TITLE
fix: usage of `a deprecated Node.js version`

### LINKED ISSUE #4060: `Test` action uses `a deprecated Node.js version`
https://github.com/jpmorganchase/salt-ds/actions/runs/10724397030

### PR DESCRIPTION
**Pull request by hamirmahal**
fixes #4060.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -111,7 +111,7 @@ jobs:
       - name: Build packages
         run: yarn run build
       - name: Cypress Components
-        uses: cypress-io/github-action@v5
+        uses: cypress-io/github-action@v6
```
Comment (hamirmahal) [inline .github/workflows/test.yml:114]: This is the latest major version at the time of this writing.

**Comment by joshwooding** [comment]
Thank you for your contribution. We require all external contributors to JPMC projects to complete our Contribution Licence Agreement (CLA). You can sign as an individual or as representing your company. You only need to do this once and it covers all current and future projects managed under our GitHub organization.

The contribution agreement can be found here [JPMC CLA](https://github.com/jpmorganchase/.github/blob/main/jpmc-cla-20230406.md?rgh-link-date=2024-07-08T16%3A03%3A16Z), follow the instructions to complete and return via email.

**Comment by hamirmahal** [comment]
You're welcome. I just sent the signed CLA via email.

**Comment by joshwooding** [comment]
> You're welcome. I just sent the signed CLA via email.

Thanks for your patience, I've just had confirmation that it's all sorted now.

**Review by joshwooding — APPROVED** [review]

**Review by mark-tate — APPROVED** [review]

**Pull request merged**